lib/licensed/sources/gradle.rb

```diff
@@ -9,127 +9,152 @@ module Licensed
       end
 
       def enumerate_dependencies
-        @dependencies ||= JSON.parse(package_metadata_command).map do |package|
+        JSON.parse(gradle_command("printDependencies", configurations: configurations)).map do |package|
           name = "#{package["group"]}:#{package["name"]}"
           Dependency.new(
             name: name,
             version: package["version"],
-            path: File.join(@config.pwd, GRADLE_LICENSES_PATH, name),
+            path: gradle_licenses_path,
+            configurations: configurations,
             metadata: {
               "type"     => Gradle.type
             }
           )
         end
       end
 
-      def with_latest_licenses
-        download_and_cache_licenses do
-          yield
-        end
-      end
-
       private
 
-      def download_licenses_from_csv
-        path = File.join(@config.pwd, GRADLE_LICENSES_PATH)
-        downloaded_licenses = {}
-        CSV.foreach(File.join(path, GRADLE_LICENSES_CSV_NAME), headers: true) do |row|
-          url = row["moduleLicenseUrl"]
-          artifact, _, version = row["artifact"].rpartition(":")
-          file_path = File.join(path, artifact, "LICENSE")
-          FileUtils.mkdir_p(File.join(path, artifact))
-          if cache_path = downloaded_licenses[url]
-            FileUtils.cp(cache_path, file_path)
+      def gradle_licenses_path
+        @gradle_licenses_path ||= File.join(config.pwd, GRADLE_LICENSES_PATH)
+      end
+
+      # Returns the configurations to include in license generation.
+      # Defaults to ["runtime", "runtimeClasspath"]
+      def configurations
+        @configurations ||= begin
+          if configurations = config.dig("gradle", "configurations")
+            Array(configurations)
           else
-            uri = URI(row["moduleLicenseUrl"])
-            File.write(file_path, Net::HTTP.get(uri))
-            downloaded_licenses[url] = file_path
+            DEFAULT_CONFIGURATIONS
           end
         end
       end
 
-      def download_and_cache_licenses
-        gradle_command("generateLicenseReport")
-        download_licenses_from_csv
-        yield
-        FileUtils.rm_rf(File.join(@config.pwd, GRADLE_LICENSES_PATH))
-      end
-
-      # Returns the output from running `npm list` to get package metadata
-      def package_metadata_command
-        gradle_command("printDependencies")
-      end
-
-      def gradle_command(*args)
-        Tempfile.open(["license", ".gradle"], @config.pwd) do |f|
-          f.write gradle_file
+      def self.gradle_command(*args, configurations:)
+        Tempfile.create do |f|
+          f.write gradle_file(configurations)
           f.close
           Licensed::Shell.execute(executable, "-q", "-b", f.path, *args)
         end
       end
 
-      def executable
-        if File.executable?(File.join(@config.pwd, "gradlew"))
+      def self.executable
+        if File.executable?(File.join(config.pwd, "gradlew"))
```

Comment:
ah, missed this one.  this would need to be cleaned up with either the path being passed in, or some other way to uniquely find the `gradlew` file